src/views/index.ts

```diff
@@ -0,0 +1,91 @@
+// Library
+import * as vscode from 'vscode';
+
+// -------
+// WEBVIEW
+// -------
+
+export class Webview {
+
+    public static readonly viewType = 'tablePreview';
+
+    /** Tracks the current panel. Only one is allowed to exist at a time */
+    public static currentPanel: Webview | undefined;
+
+    private disposables: vscode.Disposable[] = [];
+
+    private constructor(
+        private readonly panel: vscode.WebviewPanel,
+        private readonly extensionUri: vscode.Uri
+    ) {
+        // Set the webview's initial html content
+        this.update();
+
+        // Listen for when the panel is disposed
+        // This happens when the user closes the panel or when the panel is closed programmatically
+        this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
+
+        // Update the content based on the view changes
+        this.panel.onDidChangeViewState(e => {
+            if (this.panel.visible) {
+                this.update();
+            }
+        }, null, this.disposables);
+    }
+
+    private dispose() {
+        Webview.currentPanel = undefined; // Unset current panel
+        this.panel.dispose();   // Dispose off the panel
+        while (this.disposables.length) {
+            const disposable = this.disposables.pop();
+            if (disposable) {
+                disposable.dispose();
+            }
+        }
+    }
+
+    public static createOrShow(extensionUri: vscode.Uri) {
```

Comment:
Split this into two methods, `create` and `show`, and have `show` call `create` in the implementation.